.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.x
+    - name: Restore dependencies
+      run: dotnet restore GeoidHeightsDotNet.sln
+    - name: Build (debug)
+      run: dotnet build --no-restore -p:PackageVersion=$(git describe --tags) GeoidHeightsDotNet.sln
+    - name: Test
+      run: dotnet test --no-build --verbosity normal GeoidHeightsDotNet.sln
```

Comment:
I think you want to swap this order and make sure that tests are run on release build, not debug.

As-is, I *think* this is going to run the test(s) on the already built debug build.

You *could* remove the "--no-build" option to make the test step do the build also...  i think.

Reply:
> I think you want to make sure that tests are run on release build, not debug.

Why would you prefer a release build for the tests?

> As-is, I _think_ this is going to run the test(s) on the already built debug build.

Yes, that's correct (as you can see from the logs).

> You _could_ remove the "--no-build" option and add "--configuration Release" to make the test step do the release build also... i think.

I took that part from some template file. I think the idea is to separate these steps from other another, so that it's clearer where a failure happened (e.g. one could see directly in the overview that the build worked, but there was a failure in the tests).

Reply:
Sorry, I got a little off track... the main point is that I suggest making sure the tests are run on the release build, since that is what actually gets released.  I doubt there will be any difference between release and debug test runs in this simple case, but it is a good rule of thumb to test the release build, because I have seen one pass and the other fail in more complex cases.
  
The details of this script to accomplish this are up to you, building and testing in one step or build, then test.  In general, I agree that separate steps helps troubleshoot when things do go wrong, but this is a pretty simple repo, so I am tempted to combine some steps.

Reply:
@janusw  
>> Yes, that's correct (as you can see from the logs).

Where are the logs?

Reply:
> the main point is that I suggest making sure the tests are run on the release build, since that is what actually gets released. I doubt there will be any difference between release and debug test runs in this simple case, but it is a good rule of thumb to test the release build, because I have seen one pass and the other fail in more complex cases.

True. Can happen, but would hopefully be a rare case.


> And, if you really want to be paranoid, it really doesn't cost anything to test both builds in this simple repo.

Well, yeah, I'm not paranoid, but since it's cheap here, let's just do this ... ;)

Reply:
> > > Yes, that's correct (as you can see from the logs).
> 
> Where are the logs?

In this repo, they don't exist yet, since CI is not yet set up (it will be only after this PR is merged).

But you can see the logs for the experiments I did on my fork, e.g. here:
https://github.com/janusw/GeoidHeightsDotNet/runs/2134984708

Reply:
> > And, if you really want to be paranoid, it really doesn't cost anything to test both builds in this simple repo.
> 
> Well, yeah, I'm not paranoid, but since it's cheap here, let's just do this ... ;)

This is implemented in 79c17fe, so that we have a complete setup which builds and tests in debug and release mode. I hope this is ready for merging now ...